### PR TITLE
SASS 11061

### PR DESCRIPTION
**Pull request by BetaDraconis**
- **SASS-11061: Add Controller, Service, and Connector for pre-pop endpoint. Add some testing and framework classes. ITs and service/ controller test still required.**
- **SASS-11061: Adding logic and testing for detecting 'empty' CIS objects**
- **SASS-11061: Adding controller and service tests**
- **SASS-11061: Add ITs. Rename models folder for task list for consistency**

### CODE REVIEW

File: app/connectors/errors/ApiError.scala [hunk 1]
```diff
@@ -24,4 +24,6 @@ case class ApiError(status: Int, body: ApiErrorBody) {
     case error: SingleErrorBody => Json.toJson(error)
     case errors: MultiErrorsBody => Json.toJson(errors)
   }
+
+  def toLogString: String = s"with status: $status, and body: $body"
```
Comment (basotia) [inline app/connectors/errors/ApiError.scala:28]: It might be risky to log body, because of PII concerns.

**Review by basotia — APPROVED** [review]
